### PR TITLE
Fix empty body parsing

### PR DESCRIPTION
**Pull request by olegkrokhin**
When a Foundation object is destroyed and the expected HTTP response contains no body, there is currently a parsing error being thrown. There should not be an error if the body is simply empty.

Follow the QA Plan in https://github.com/sportngin/ngin/pull/7939

### CODE REVIEW

File: spec/typhoid/parser_spec.rb [hunk 1]
```diff
@@ -53,6 +55,10 @@ module Typhoid
         it "has an expected element" do
           subject.parse["result"]["type"].should == "orphan"
         end
+
+        it "has no body" do
+          expect(Parser.new(empty_body).parse).to be_nil
+        end
```
Comment (kjbkjb) [inline spec/typhoid/parser_spec.rb]: maybe add 
```
it "empty json body should not raise" do
  exect{Parser.new(empty_body)}.to_not raise_error ...
end
```

^ Just pseudo-code

**Comment by olegkrokhin** [comment]
@kjbkjb Any other comments/changes?

**Comment by kjbkjb** [comment]
CR :+1:

**Comment by melcollova** [comment]
@olegkrokhin QA :+1: Per sportngin/ngin#7939

**Pull request merged**